lib/puppet/provider/ssh_authorized_key/parsed.rb

```diff
@@ -56,16 +56,10 @@ def flush
     # so calling it here suppresses the later attempt by our superclass's flush method.
     self.class.backup_target(target)
 
-    Puppet::Util::SUIDManager.asuser(@resource.should(:user)) do
-      unless Puppet::FileSystem.exist?(dir = File.dirname(target))
-        Puppet.debug "Creating #{dir} as #{@resource.should(:user)}"
-        Dir.mkdir(dir, dir_perm)
-      end
+    super
 
-      super
-
-      File.chmod(file_perm, target)
-    end
+    File.chmod(file_perm, target)
+    FileUtils.chown(@resource.should(:user), @resource.should(:user), target)
```

Comment:
second argument should be group? => FileUtils.chown(@resource.should(:user), @resource.should(:group), target)

Reply:
Group is not a property we have for the `ssh_authorized_key` type